skynet.go

```diff
@@ -8,62 +8,85 @@ import (
 	"mime/multipart"
 	"net/http"
 	"os"
+	gopath "path"
 	"path/filepath"
 	"strings"
+
+	"gitlab.com/NebulousLabs/errors"
 )
 
 type (
-	UploadReponse struct {
-		Skylink string `json:"skylink"`
-	}
-
+	// UploadOptions contains the options used for uploads.
 	UploadOptions struct {
-		portalUrl                    string
-		portalUploadPath             string
-		portalFileFieldname          string
-		portalDirectoryFileFieldname string
-		customFilename               string
+		// PortalURL is the URL of the portal to use.
+		PortalURL string
+		// PortalUploadPath is the path on the portal of the upload endpoint.
```

Comment:
This is not very clear to me. Is this the local path on the server where the files will be stored or the relative URL path of the endpoint?